packages/SwingSet/src/vats/comms/delivery.js

```diff
@@ -367,8 +432,9 @@ export function makeDeliveryKit(state, syscall, transmit, clistKit, stateKit) {
   function resolveToKernel(localResolutions) {
     const resolutions = [];
     for (const localResolution of localResolutions) {
-      const [vpid, rejected, data] = localResolution;
-      resolutions.push([vpid, rejected, mapDataToKernel(data)]);
+      const [lpid, rejected, data] = localResolution;
+      const kfpid = provideKernelForLocal(lpid);
```

Comment:
also pretty sure this shouldn't be allowed to allocate

Reply:
I'm pretty sure I _do_ hear an echo.